pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java

```diff
@@ -151,18 +153,32 @@ public void start() {
         runnerThread.start();
     }
 
-    public abstract V extractValue(ConsumerRecord<String, byte[]> record);
+    public Object extractValue(ConsumerRecord<Object, Object> consumerRecord) {
+        return consumerRecord.value();
+    }
 
+    public Optional<String> extractKey(ConsumerRecord<Object, Object> consumerRecord) {
+        // we are currently supporting only String keys
+        return Optional.ofNullable((String) consumerRecord.key());
```

Comment:
since the key is always a String, would it make sense to set that already in the ConsumerRecord type parameter?

Reply:
This is only a transient situation.
Because we want to prepare the way to deal with non String keys.
IIRC is was a @sijie's request to introduce this 'extractKey' method in order to make it clear the direction we are taking

Reply:
Wouldn't it be cleaner to simply handle what is currently supported? One reason could be to address the compiler warnings related to unchecked or unsafe operations without suppressing the compiler warnings. If compiler warnings are ignored, it probably doesn't make a difference.